### PR TITLE
LPS-187443 Add POSHI automation | Master

### PR DESCRIPTION
**Pull request by alessandromalves**
This is only for master
https://issues.liferay.com/browse/LPS-187443
cc.: @david-gutierrez-mesa @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by alessandromalves** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase [hunk 1]
```diff
@@ -992,6 +992,53 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-180106. User is Able to edit Navigation Menu even if another portal instance contains a custom object and database partitioning is enabled."
+	@priority = 4
+	test EditNavigationMenuInVirtualInstanceWhenDBPartitioningEnabled {
+		property custom.properties = "include-and-override=portal-liferay-online.properties${line.separator}virtual.hosts.valid.hosts=localhost,127.0.0.1,www.able.com";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:998]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase [hunk 2]
```diff
@@ -992,6 +992,53 @@ definition {
...
+		task ("When a new Virtual Instance is created") {
+			HeadlessPortalInstanceAPI.addPortalInstance(
+				domain = "www.able.com",
+				portalInstanceId = "www.able.com",
+				virtualHost = "www.able.com");
+
+			User.firstLoginUI(
+				password = "test",
+				specificURL = "http://www.able.com:8080",
+				userEmailAddress = "test@www.able.com");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:1028]:
```
			HeadlessPortalInstanceAPI.addPortalInstance(
				domain = "liferay.com",
				portalInstanceId = "www.able.com",
				virtualHost = "www.able.com");

			User.firstLoginPG(
				password = "test",
				userEmailAddress = "test@liferay.com",
				virtualHostsURL = "http://www.able.com:8080");

```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase [hunk 3]
```diff
@@ -992,6 +992,53 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-180106. User is Able to edit Navigation Menu even if another portal instance contains a custom object and database partitioning is enabled."
+	@priority = 4
+	test EditNavigationMenuInVirtualInstanceWhenDBPartitioningEnabled {
+		property custom.properties = "include-and-override=portal-liferay-online.properties${line.separator}virtual.hosts.valid.hosts=localhost,127.0.0.1,www.able.com";
+		property database.partition.enabled = "true";
+		property test.liferay.virtual.instance = "false";
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:1001]: var labelName = StringUtil.randomString(8);

		var objectName = "A${labelName}";

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase [hunk 4]
```diff
@@ -992,6 +992,53 @@ definition {
...
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Field",
+				fieldName = "customObjectField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "TestCustomObject");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "TestCustomObject");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:1016]:
```
			ObjectAdmin.addObjectViaAPI(
				labelName = ${labelName},
				objectName = ${objectName},
				pluralLabelName = "${labelName}s");

			ObjectAdmin.addObjectFieldViaAPI(
				fieldBusinessType = "Text",
				fieldLabelName = "Name",
				fieldName = "name",
				fieldType = "String",
				isRequired = "false",
				objectName = ${objectName});

			ObjectAdmin.publishObjectViaAPI(objectName = ${objectName});

```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase [hunk 5]
```diff
@@ -992,6 +992,53 @@ definition {
...
+				domain = "www.able.com",
+				portalInstanceId = "www.able.com",
+				virtualHost = "www.able.com");
+
+			User.firstLoginUI(
+				password = "test",
+				specificURL = "http://www.able.com:8080",
+				userEmailAddress = "test@www.able.com");
+
+			Navigator.openSpecificURL(url = "http://www.able.com:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_site_navigation_admin_web_portlet_SiteNavigationAdminPortlet");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:1030]:
```
			ProductMenu.gotoPortlet(
				category = "Site Builder",
				portlet = "Navigation Menus");

```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase [hunk 6]
```diff
@@ -992,6 +992,53 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-180106. User is Able to edit Navigation Menu even if another portal instance contains a custom object and database partitioning is enabled."
+	@priority = 4
+	test EditNavigationMenuInVirtualInstanceWhenDBPartitioningEnabled {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:997]: EditNavigationMenuInVirtualInstanceWithDBPartitioning

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase (hunk 3 above)
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:1001]: property database.types = "mysql";

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase [hunk 7]
```diff
@@ -992,6 +992,53 @@ definition {
...
+		property custom.properties = "include-and-override=portal-liferay-online.properties${line.separator}virtual.hosts.valid.hosts=localhost,127.0.0.1,www.able.com";
+		property database.partition.enabled = "true";
+		property test.liferay.virtual.instance = "false";
+
+		task ("Given portal instance contains a custom object and database partitioning is enabled") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:1002]: Given a user has a custom object in default instance

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase [hunk 8]
```diff
@@ -992,6 +992,53 @@ definition {
...
+				fieldLabelName = "Field",
+				fieldName = "customObjectField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "TestCustomObject");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "TestCustomObject");
+		}
+
+		task ("When a new Virtual Instance is created") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:1019]: When the user adds a navigation menu in virtual instance

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase [hunk 9]
```diff
@@ -992,6 +992,53 @@ definition {
...
+				password = "test",
+				specificURL = "http://www.able.com:8080",
+				userEmailAddress = "test@www.able.com");
+
+			Navigator.openSpecificURL(url = "http://www.able.com:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_site_navigation_admin_web_portlet_SiteNavigationAdminPortlet");
+
+			NavigationMenusAdmin.addMenu(menuName = "Test Navigation Menu Name");
+		}
+
+		task ("Then the navigation menu editor should load properly") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:1035]: Then the user could edit the navigation menu.

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @alessandromalves, could you modify according to comments? Thanks

**Comment by alessandromalves** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
LGTM! @alessandromalves

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**